frame/staking/src/tests.rs

```diff
@@ -5287,3 +5292,160 @@ fn proportional_ledger_slash_works() {
+			Staking::ledger(&4).unwrap(),
+			StakingLedger {
+				stash: 3,
+				total: 1500,
+				active: 1500,
+				unlocking: Default::default(),
+				claimed_rewards,
+			}
+		);
+
+		// start next era
+		current_era = current_era + 1;
+		mock::start_active_era(current_era);
+
+		// claiming reward for last era in which validator was active works
+		assert_ok!(Staking::payout_stakers(Origin::signed(4), 3, current_era - 1));
+
+		// consumed weight for all payout_stakers dispatches that fail
+		let err_weight = weights::SubstrateWeight::<Test>::payout_stakers_alive_staked(0);
+		// cannot claim rewards for an era before bonding occured as it is
+		// already marked as claimed.
+		assert_noop!(
+			Staking::payout_stakers(Origin::signed(4), 3, current_era - 2),
+			Error::<Test>::AlreadyClaimed.with_weight(err_weight)
+		);
+
+		// decoding will fail now since Staking Ledger is in corrupt state
+		HistoryDepth::set(history_depth - 1);
+		assert_eq!(Staking::ledger(&4), None);
+
+		// make sure stakers still cannot claim rewards that they are not meant to
+		assert_noop!(
+			Staking::payout_stakers(Origin::signed(4), 3, current_era - 2),
+			Error::<Test>::NotController
+		);
+
+		// fix the corrupted state for post conditions check
+		HistoryDepth::set(history_depth);
+	});
+}
+
+#[test]
+fn reducing_history_depth_without_migration() {
+	// Verifies initial conditions of mock
+	ExtBuilder::default().nominate(false).build_and_execute(|| {
+		let original_history_depth = HistoryDepth::get();
+		let mut current_era = original_history_depth + 10;
+		let last_reward_era = current_era - 1;
+		let start_reward_era = current_era - original_history_depth;
+
+		// put some money in (stash, controller)=(3,4),(5,6).
+		for i in 3..7 {
+			let _ = Balances::make_free_balance_be(&i, 2000);
+		}
+
+		// start current era
+		mock::start_active_era(current_era);
+
+		// add a new candidate for being a staker. account 3 controlled by 4.
+		assert_ok!(Staking::bond(Origin::signed(3), 4, 1500, RewardDestination::Controller));
+
+		// all previous era before the bonding action should be marked as
+		// claimed.
+		let mut claimed_rewards = vec![];
+		for i in start_reward_era..=last_reward_era {
+			claimed_rewards.push(i);
+		}
+
+		let claimed_rewards: BoundedVec<_, _> = claimed_rewards.try_into().unwrap();
+		assert_eq!(
+			Staking::ledger(&4).unwrap(),
+			StakingLedger {
+				stash: 3,
+				total: 1500,
+				active: 1500,
+				unlocking: Default::default(),
+				claimed_rewards,
+			}
+		);
+
+		// next era
+		current_era = current_era + 1;
+		mock::start_active_era(current_era);
+
+		// claiming reward for last era in which validator was active works
+		assert_ok!(Staking::payout_stakers(Origin::signed(4), 3, current_era - 1));
+
+		// next era
+		current_era = current_era + 1;
+		mock::start_active_era(current_era);
+
+		// history_depth reduced without migration
+		let history_depth = original_history_depth - 1;
+		HistoryDepth::set(history_depth);
```

Comment:
```suggestion
		HistoryDepth::mutate(|x| *x -= 1);
```

I think we also implement this.

Reply:
I need to keep track of `history_depth` (current) for the assertions after this and `original_history_depth` to fix the corrupt state in order for the `post_conditions()` of the test to pass.